paparazzi/src/main/java/app/cash/paparazzi/internal/ImageUtils.kt

```diff
@@ -312,6 +313,40 @@ internal object ImageUtils {
     }
   }
 
+  fun smallestDiffRect(firstImage: BufferedImage, secondImage: BufferedImage): Rectangle? {
+    var rect: Rectangle? = null
+    val width = max(firstImage.width, secondImage.width)
+    val height = max(firstImage.height, secondImage.height)
+
+    for (y in 0 until height) {
+      for (x in 0 until width) {
+        val firstRgb = if (x < firstImage.width && y < firstImage.height) firstImage.getRGB(x, y) else 0
+        val secondRgb = if (x < secondImage.width && y < secondImage.height) secondImage.getRGB(x, y) else 0
+        if (firstRgb != secondRgb) {
+          rect = if (rect == null) {
+            Rectangle(x, y, 1, 1)
+          } else {
+            rect.union(Rectangle(x, y, 1, 1))
+          }
+        }
+      }
+    }
+
+    return rect
+  }
+
+  fun BufferedImage.resize(targetWidth: Int, targetHeight: Int): BufferedImage {
+    if (width == targetWidth && height == targetHeight) {
+      return this
+    }
+
+    return BufferedImage(targetWidth, targetHeight, type).apply {
+      val g = graphics
```

Comment:
I think you want this for better image quality?
```
 g.setRenderingHint(RenderingHints.KEY_INTERPOLATION,
                RenderingHints.VALUE_INTERPOLATION_BICUBIC);
```

Reply:
```suggestion
      val g = createGraphics()
```

Reply:
Should I just call `setRenderingHints`?

```
  private fun setRenderingHints(g2: Graphics2D) {
    g2.setRenderingHint(KEY_INTERPOLATION, VALUE_INTERPOLATION_BILINEAR)
    g2.setRenderingHint(KEY_RENDERING, VALUE_RENDER_QUALITY)
    g2.setRenderingHint(KEY_ANTIALIASING, VALUE_ANTIALIAS_ON)
  }
```

Reply:
Actually I don't think we need to do this, we are doing a direct pixel copy into the resulting image with no scaling or anything that would benefit from hints